GVFS/GVFS.Service/GVFSMountProcess.Mac.cs

```diff
@@ -1,52 +1,72 @@
 ﻿using GVFS.Common;
-using GVFS.Common.Tracing;
-using GVFS.Service.Handlers;
-using System;
+using GVFS.Common.Tracing;
+using System.Diagnostics;
+using System.IO;
 
 namespace GVFS.Service
 {
-    public class GVFSMountProcess : IDisposable
+    public class GVFSMountProcess : IRepoMounter
     {
-        private const string ParamPrefix = "--";
+        private const string ExecutablePath = "/bin/launchctl";
 
-        private readonly ITracer tracer;
+        private MountLauncher processLauncher;
+        private GVFSPlatform platform;
 
-        public GVFSMountProcess(ITracer tracer, int sessionId)
+        public GVFSMountProcess(MountLauncher processLauncher = null, GVFSPlatform platform = null)
         {
-            this.tracer = tracer;
+            this.processLauncher = processLauncher ?? new MountLauncher();
+            this.platform = platform ?? GVFSPlatform.Instance;
         }
 
-        public bool Mount(string repoRoot)
+        public bool MountRepository(string repoRoot, int sessionId, ITracer tracer)
         {
-            if (!this.CallGVFSMount(repoRoot))
+            string arguments = string.Format(
+                "asuser {0} {1} mount {2}",
+                sessionId,
+                Path.Combine(this.platform.Constants.GVFSBinDirectoryPath, this.platform.Constants.GVFSExecutableName),
+                repoRoot);
+
+            if (!this.processLauncher.LaunchProcess(ExecutablePath, arguments, repoRoot, tracer))

```

Comment:
@alameenshah had the idea to use an out-param here to avoid multiple trace output.

Reply:
It turned out to be easier to move tracer to the Constructor.